soc/xtensa/intel_s1000/include/_soc_inthandlers.h

```diff
@@ -0,0 +1,257 @@
+/*
+ * THIS FILE WAS AUTOMATICALLY GENERATED.  DO NOT EDIT.
+ *
+ * Functions here are designed to produce efficient code to
+ * search an Xtensa bitmask of interrupts, inspecting only those bits
+ * declared to be associated with a given interrupt level.  Each
+ * dispatcher will handle exactly one flagged interrupt, in numerical
+ * order (low bits first) and will return a mask of that bit that can
+ * then be cleared by the calling code.  Unrecognized bits for the
+ * level will invoke an error handler.
```

Comment:
This reminds me: I didn't have enough cmake-fu to get the python script to generate these from the core-isa.h at build time, which is why these files are checked into git.  I bet I could do it now...